### PR TITLE
Mark System.Runtime APIs Supported on Windows

### PR DESCRIPTION
**Pull request by mdh1418**
```
all mono M:System.ArgIterator.#ctor(System.RuntimeArgumentHandle),System,ArgIterator,.ctor(RuntimeArgumentHandle),0
all mono "M:System.ArgIterator.#ctor(System.RuntimeArgumentHandle,System.Void*)",System,ArgIterator,".ctor(RuntimeArgumentHandle, Void*)",0
serialization "M:System.Delegate.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)",System,Delegate,"GetObjectData(SerializationInfo, StreamingContext)",0
all PNSE M:System.Exception.remove_SerializeObjectState(System.EventHandler{System.Runtime.Serialization.SafeSerializationEventArgs}),System,Exception,remove_SerializeObjectState(EventHandler<SafeSerializationEventArgs>),0
all PNSE M:System.Exception.add_SerializeObjectState(System.EventHandler{System.Runtime.Serialization.SafeSerializationEventArgs}),System,Exception,add_SerializeObjectState(EventHandler<SafeSerializationEventArgs>),0
all mono M:System.GC.GetGCMemoryInfo(System.GCKind),System,GC,GetGCMemoryInfo(GCKind),0
serialization "M:System.RuntimeFieldHandle.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)",System,RuntimeFieldHandle,"GetObjectData(SerializationInfo, StreamingContext)",0
serialization "M:System.RuntimeMethodHandle.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)",System,RuntimeMethodHandle,"GetObjectData(SerializationInfo, StreamingContext)",0
serialization "M:System.RuntimeTypeHandle.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)",System,RuntimeTypeHandle,"GetObjectData(SerializationInfo, StreamingContext)",0
Partially PNSE M:System.String.IsNormalized(System.Text.NormalizationForm),System,String,IsNormalized(NormalizationForm),3
Partially PNSE M:System.String.Normalize(System.Text.NormalizationForm),System,String,Normalize(NormalizationForm),3
Not implemented by design M:System.Type.GetType(System.String),System,Type,GetType(String),3
Not implemented by design "M:System.Type.GetType(System.String,System.Boolean,System.Boolean)",System,Type,"GetType(String, Boolean, Boolean)",3
all mono "M:System.Type.GetTypeFromCLSID(System.Guid,System.String,System.Boolean)",System,Type,"GetTypeFromCLSID(Guid, String, Boolean)",0
all mono "M:System.Type.GetTypeFromProgID(System.String,System.String,System.Boolean)",System,Type,"GetTypeFromProgID(String, String, Boolean)",0
all mono M:System.Type.GetTypeFromProgID(System.String),System,Type,GetTypeFromProgID(String),1
all mono "M:System.Type.GetTypeFromProgID(System.String,System.Boolean)",System,Type,"GetTypeFromProgID(String, Boolean)",1
all mono M:System.Type.GetTypeFromCLSID(System.Guid),System,Type,GetTypeFromCLSID(Guid),1
all mono "M:System.Type.GetTypeFromProgID(System.String,System.String)",System,Type,"GetTypeFromProgID(String, String)",1
all mono "M:System.Type.GetTypeFromCLSID(System.Guid,System.Boolean)",System,Type,"GetTypeFromCLSID(Guid, Boolean)",1
all PNSE "M:System.Type.ReflectionOnlyGetType(System.String,System.Boolean,System.Boolean)",System,Type,"ReflectionOnlyGetType(String, Boolean, Boolean)",0
Not implemented by design "M:System.Type.GetType(System.String,System.Boolean)",System,Type,"GetType(String, Boolean)",3
all mono "M:System.Type.GetTypeFromCLSID(System.Guid,System.String)",System,Type,"GetTypeFromCLSID(Guid, String)",1
```

Of the APIs listed to be marked as unsupported, many were serialization related (had `System.Runtime.Serialization.SerializationInfo` as a parameter), throw PNSE everywhere, or throw PNSE on mono.

This PR marks `System.Type.GetTypeFromCLSID` and `System.Type.GetTypeFromProgID` as Supported on Windows only.

### CODE REVIEW

**Comment by mdh1418** [comment]
The remaining methods seem like false-positives, but I wasn't sure. @MaximLipnin @steveisok could y'all take a look?

**Comment by jkotas** [comment]
> The remaining methods seem like false-positives, 

Agree.

**Comment by jkotas** [comment]
> System.Type.GetTypeFromCLSID
> System.Type.GetTypeFromProgID

It may make sense to mark these as Windows-specific.

**Comment by mdh1418** [comment]
Both `System.String.IsNormalized(System.Text.NormalizationForm)` and `System.String.Normalize(System.Text.NormalizationForm)` work on Browser except for when NormalizationForm is `NormalizationForm.FormKC` or `NormalizationForm.FormKD`. 
https://github.com/dotnet/runtime/blob/8fc68f626a11d646109a758cb0fc70a0aa7826f1/src/libraries/System.Private.CoreLib/src/System/Globalization/Normalization.Icu.cs#L104.

Since the method is partially supported, what should be done in this case?

File: src/libraries/System.Private.CoreLib/src/System/Type.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 using System.Reflection;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
```
Comment (safern) [inline src/libraries/System.Private.CoreLib/src/System/Type.cs:10]: Unintentional change?

Comment (mdh1418) [reply src/libraries/System.Private.CoreLib/src/System/Type.cs:10]: Yeah :) pushing new changes soon.

**Review by marek-safar — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Pull request merged**